pallets/services-payment/src/lib.rs

```diff
@@ -114,6 +114,11 @@ pub mod pallet {
     pub type BlockProductionCredits<T: Config> =
         StorageMap<_, Blake2_128Concat, ParaId, BlockNumberFor<T>, OptionQuery>;
 
+    /// List of para ids that have already been given free credits
+    #[pallet::storage]
+    #[pallet::getter(fn given_free_credits)]
+    pub type GivenFreeCredits<T: Config> = StorageMap<_, Blake2_128Concat, ParaId, (), OptionQuery>;
```

Comment:
Add a few tests for this:

* After migration, calling mark_valid_for_collating on a paused para id does not give it more free credits
* Register, mark_valid_for_collating, deregister, register again, mark_valid_for_collating again, should not receive free credits the 2nd time

Reply:
pretty much yeah, also if I inject my own free-credits I at most get the max for free (it does not add up)

Reply:
As always, I would add a sudo extrinsic to clean this storage

Reply:
All done